### PR TITLE
Improve SplitContainer behavior, keeping asked split position whenever possible

### LINKED ISSUE #43749: Custom dock with minimal size appears bigger than necessary
**Godot version:**

3.2.4 beta2

**Issue description:**

uyBuazcFLC
The dock has minimum horizontal size. It looks like the panel is resized to this size, but it's also added to the final size, so the left panel is wider than necessary. But when you manually shrink the width to minimal, when the dock is switched out the additional size is still subtracted resulting in tiny panel.

**Minimal reproduction project:**

[ReproductionProject.zip](https://github.com/godotengine/godot/files/5577793/ReproductionProject.zip)

### PR DESCRIPTION
**Pull request by groud**
Fixes https://github.com/godotengine/godot/issues/43749. (edited with the correct issue number)

### CODE REVIEW

**Comment by KoBeWi** [comment]
This fixes #43749 not #64273
While the latter is very similar, it's also about Import dock being too wide.

File: scene/gui/split_container.cpp [hunk 1]
```diff
@@ -55,58 +55,71 @@ Control *SplitContainer::_getch(int p_idx) const {
...
-			fit_child_in_rect(second, Rect2(Point2(), get_size()));
-		}
-		return;
-	}
-
 	// Determine expanded children
 	bool first_expanded = (vertical ? first->get_v_size_flags() : first->get_h_size_flags()) & SIZE_EXPAND;
 	bool second_expanded = (vertical ? second->get_v_size_flags() : second->get_h_size_flags()) & SIZE_EXPAND;
 
+	// Compute the minimum size
```
Comment (KoBeWi) [inline scene/gui/split_container.cpp:66]:
```suggestion
	// Compute the minimum size.
```
If you move a comment, you have to fix it. Such are the rules :v
(same below)

**Review by KoBeWi — APPROVED** [review]
I tested different cases and it seems to work well; can't say much about the code.

CC @Rindbee as it supersedes your PR.

File: scene/gui/split_container.cpp (hunk 1 above)
Comment (groud) [reply scene/gui/split_container.cpp:66]: Alright alright! :smile:

**Comment by Rindbee** [comment]
`split_offset` seems to be calculated relative to the container. The origin is no longer dynamic, unless both expands are enabled.

**Comment by groud** [comment]
> split_offset seems to be calculated relative to the container. The origin is no longer dynamic, unless both expands are enabled.

Yeah, that's what solve the issue. I think it's the most expected behavior, but indeed, it kind of breaks compatibility.

**Pull request merged**